lib/common/http1client.c

```diff
@@ -576,27 +576,33 @@ static h2o_iovec_t build_request(struct st_h2o_http1client_t *client, h2o_iovec_
 static void on_connection_ready(struct st_h2o_http1client_t *client)
 {
     h2o_iovec_t proxy_protocol = h2o_iovec_init(NULL, 0);
-    int chunked = 0;
     h2o_iovec_t connection_header = h2o_iovec_init(NULL, 0);
     h2o_httpclient_properties_t props = {
-        &proxy_protocol,
-        &chunked,
-        &connection_header,
+        &proxy_protocol, &connection_header
     };
     h2o_iovec_t method;
     h2o_url_t url;
     h2o_header_t *headers;
     size_t num_headers;
-    h2o_iovec_t body;
+    h2o_httpclient_req_body_t body = {H2O_HTTPCLIENT_REQ_BODY_NONE};
 
     client->super._cb.on_head = client->super._cb.on_connect(&client->super, NULL, &method, &url, (const h2o_header_t **)&headers,
-                                                             &num_headers, &body, &client->proceed_req, &props, client->_origin);
+                                                             &num_headers, &body, &props, client->_origin);
 
     if (client->super._cb.on_head == NULL) {
         close_client(client);
         return;
     }
 
+    {
+        h2o_headers_t headers_vec = (h2o_headers_t){headers, num_headers, num_headers};
+        int chunked = 0;
+        h2o_httpclient__add_cl_or_te_header(client->super.pool, method, &headers_vec, &body, &chunked);
+        headers = headers_vec.entries;
+        num_headers = headers_vec.size;
+        client->_is_chunked = chunked;
+    }
```

Comment:
This looks to me like a hack. Can't we instead do the following?
* step 1. ask h2o_httpclient what values needs to be sent (i.e., the value in a number of content-length, the value of the transfer-encoding header)
* step 2. supply that to the `build_request` function as extra arguments, like `*props.connection_header`